mypy/fastparse.py

```diff
@@ -213,10 +214,56 @@ def parse(
         assert options.python_version[0] >= 3
         feature_version = options.python_version[1]
     try:
+        ignore_comment_errors = options and options.ignore_comment_errors
+
+        def try_parse_ast(source: str) -> tuple[AST | None, str]:
+            try:
+                ast = ast3_parse(source, fnam, "exec", feature_version=feature_version)
+            except SyntaxError as e:
+
+                def strip_comment(line: str) -> str:
+                    return re.sub(r"([^#]*).*(\r\n|\r|\n)?", r"\1\2", line)
+
+                def strip_comment_from_line(text: str, target_lineno: int) -> str:
+                    lines = text.splitlines(keepends=True)
+                    return "".join(
+                        line if lineno != target_lineno else strip_comment(line)
+                        for lineno, line in enumerate(lines, start=1)
+                    )
+
+                if (
+                    ignore_comment_errors
+                    and e.text
+                    and e.lineno
+                    and TYPE_COMMENT_PATTERN.match(e.text)
+                ):
+                    stripped_source = strip_comment_from_line(source, e.lineno)
+                    errors.report(
+                        e.lineno or -1,
+                        e.offset or -1,
+                        # TODO: It would be nicer if the message included the line text via
+                        # `e.text.strip()`. However, I can't figure out how to make the unit
+                        # test pass, because the # N: <expected_result> in the unit test must
+                        # contain itself. If there was some partial matching mechanism for the
+                        # unit test comment, then we could make this work.
+                        "Ignored bad type comment",
+                        severity="note",
+                        code=codes.SYNTAX,
+                    )
+                    return None, stripped_source
+                raise
+            return ast, source
+
         # Disable deprecation warnings about \u
         with warnings.catch_warnings():
             warnings.filterwarnings("ignore", category=DeprecationWarning)
-            ast = ast3_parse(source, fnam, "exec", feature_version=feature_version)
+            ast: AST | None = None
+            str_source = source if isinstance(source, str) else source.decode()
```

Comment:
`source` isn't always a `str`?

Reply:
According to the type annotation, source is str | bytes. In fact, a few py38 unit tests (that are skipped in newer versions) fail because they assert error on bad encodings for byte sources. Line 261 masks this problem by assuming the encoding of bytes sources is always UTF-8. I just pushed a commit to the fork branch that fixes that issue. Do I have to do anything to this pull request to incorporate that commit, or does it show up here automatically?

Reply:
Ah, never mind 🤦  Could've noticed it myself if GitHub defaulted to showing more context...

re
> Do I have to do anything to this pull request to incorporate that commit, or does it show up here automatically?

a pull request is automatically updated to show whatever you push to its branch